theano/gof/utils.py

```diff
@@ -210,6 +213,15 @@ def __str__(self):
 
         return type.__new__(cls, name, bases, dct)
 
+    _cache = {}
+
+    def __call__(cls, *args, **kwargs):
+        if not cls._has_props:
```

Comment:
@Sentient07 this "if" will make this method have as the default when there is no \__props__. So it support class that don't have problems by not caching them. This is what we want.